# 0. Data Scrapping

In this simple notebook, we scrap the Lovecraft texts data from the GitHub repository $\tt {https://github.com/vilmibm/lovecraftcorpus}$. We thank user vilmibm (Nate Smith) for collecting Lovecraft's works in $\tt .txt$ form, which was very convenient for the analysis.

In [ ]:
# Libraries

# File management
from pathlib import Path                  # imports path tools from Pathlib for working with file paths
import os                                 # imports the OS library for interacting with the operating system
import shutil                             # imports the Shutil library for high-level file operations

# defines a function to find the project root directory by looking for a specific marker (default is 'data')
def find_project_root(start=Path.cwd(), marker='data'): 
    current = start
    while current != current.parent:
        if (current / marker).exists():
            return current
        current = current.parent
    raise FileNotFoundError("Project root not found")

# chaning directory to the bronze data folder
PROJECT_ROOT = find_project_root()                   # obtains the project root
BRONZE_ROOT = str(PROJECT_ROOT) + '/data/bronze'     # bronze data folder path
os.chdir(BRONZE_ROOT)                                # changes the dictory to the bronze data folder


We now proceed to clone the directory and rename it to $\tt 'texts'$. 

In [ ]:
# texts path variable
texts_path =  str(BRONZE_ROOT) + '/texts' 

# deletes the texts directory if it exists
if os.path.exists(texts_path):
    shutil.rmtree(texts_path)

# clones the repository 
os.system("git clone https://github.com/vilmibm/lovecraftcorpus")

# removing git files from repository
os.remove("lovecraftcorpus/README.md")
#os.remove("lovecraftcorpus/.gitignore")
shutil.rmtree("lovecraftcorpus/.git")

# renames the folder to 'texts'
os.rename("lovecraftcorpus", "texts")

Cloning into 'lovecraftcorpus'...


We now make a simple inspection of the number $N$ of texts to estimate the precision of the statistics that we are going to extract through the relation $\sigma \approx 1/\sqrt{N}$.

In [20]:
# number of files in the text folder
N = len(os.listdir(texts_path) )

print('   N = ' + str(N) + ' texts')
print( r'=> σ = 1/√N ≅ ' + f"{1/N**(1/2):.3f}" )

   N = 67 texts
=> σ = 1/√N ≅ 0.122


We then observe that our precision will be of relative order $\sigma \approx 0.122$.